### DISCUSSION TITLE
Export result set from temp tables error

### DISCUSSION THREAD

**Discussion by ems2811** [dbeaver/dbeaver#15452] [Q&A] [answered]
#### System information: 
- Win10 Pro
- DBeaver 21.3.4 (same problem in 21.3.3)
- No additional extensions

#### Connection specification:
- PostgreSQL 9.5
- Default postgresql driver
- No tunnels or proxies

#### Describe the problem you're observing:
Error on export result set from temp table

#### Steps to reproduce, if exist:
Execute query below and try export to SQL or TXT format
```
create temp table tt as
select 1 as id;

select *
from tt;
```
![01](https://user-images.githubusercontent.com/22142845/152980971-04d0d3ca-81f9-4d98-8414-7e3868c3f675.jpg)

#### Include any warning/errors/backtraces from the logs
[log](https://github.com/dbeaver/dbeaver/files/8023482/log.log)
[dbeaver-debug.log](https://github.com/dbeaver/dbeaver/files/8023485/dbeaver-debug.log)

**Comment by uslss** [comment, marked as answer]
@ems2811 probably, you've got Open new connection(s) enabled in properties?
![image](https://user-images.githubusercontent.com/31996417/153060775-1091bbfd-1b3a-4d50-86fe-1a2cc12d0934.png)
In this case the temporary table doesn't exist in the newly opened connection

**Comment by ems2811** [comment]
Really. I didn't think about that. Thank you.

**Comment by dstj** [comment]
@uslss 

I don't know if this is by SQL design or a bug in DBeaver v24.2.3, but it does not work when the whole thing is running inside a transaction. Well, it works only once...

**Reproduction**
```sql
BEGIN TRANSACTION;

CREATE TEMP TABLE tmp1 AS
SELECT 1;

CREATE TEMP TABLE tmp2 AS
SELECT 2;

SELECT * FROM tmp1;
SELECT * FROM tmp2;

-- ROLLBACK;
```
_(Please note that the ROLLBACK is commented out (I'm running a migration test).)_

Exporting the results of the first `tmp1` works, then exporting the results of `tmp2` errors with:
```
SQL Error [42P01]: ERROR: relation "tmp2" does not exist
Position: 15
```

Trying to export the `tmp1` results a second time also yields the error mind you.

Do you know if that is expected or a bug? I really need to find a way to export the results of my 8-9 temp tables during this data migration.

Thanks.
